File: api/Entities/Contract.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace RentalControl.Entities;

[Table("Contracts")]
public class Contract : BaseModel
{
    [PrimaryKey] public Guid Id { get; set; }
    [Column] public Guid TenantId { get; set; }
    [Column] public Guid AddressId { get; set; }
    [Column] public decimal Deposit { get; set; }
    [Column] public decimal Rent { get; set; }
    [Column] public string Business { get; set; } = string.Empty;
    [Column] public DateTime StartDate { get; set; }
    [Column] public DateTime EndDate { get; set; }
    [Column] public ContractStatus Status { get; set; }
    [Column] public ContractType Type { get; set; }
    [Column] public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Reference(typeof(Tenant))] public Tenant Tenant { get; set; } = new();
    [Reference(typeof(Address))] public Address Address { get; set; } = new();
    [Reference(typeof(ContractGuarantor))] public List<Guarantor> Guarantors { get; set; } = [];
}

public enum ContractStatus
{
    Active,
    Expired,
    Terminated
}

public enum ContractType
{
    Yearly
}

File: api/dto/contract.go
package dto

import (
	"github.com/google/uuid"
	"time"
)

type CreateContractRequest struct {
	TenantID     uuid.UUID   `json:"tenant_id" binding:"required"`
	AddressID    uuid.UUID   `json:"address_id" binding:"required"`
	Deposit      float64     `json:"deposit" binding:"required"`
	Rent         float64     `json:"rent" binding:"required"`
	Business     string      `json:"business" binding:"required"`
	StartDate    time.Time   `json:"start_date" binding:"required"`
	EndDate      time.Time   `json:"end_date" binding:"required"`
	Status       string      `json:"status" binding:"required"`
	Type         string      `json:"type" binding:"required"`
	GuarantorIDs []uuid.UUID `json:"guarantor_ids"`
}

type UpdateContractRequest struct {
	ID           uuid.UUID   `json:"id" binding:"required"`
	TenantID     uuid.UUID   `json:"tenant_id"`
	AddressID    uuid.UUID   `json:"address_id"`
	Deposit      float64     `json:"deposit"`
	Rent         float64     `json:"rent"`
	Business     string      `json:"business"`
	StartDate    time.Time   `json:"start_date"`
	EndDate      time.Time   `json:"end_date"`
	Status       string      `json:"status"`
	Type         string      `json:"type"`
	GuarantorIDs []uuid.UUID `json:"guarantor_ids"`
}

type ContractResponse struct {
	ID         uuid.UUID           `json:"id"`
	TenantID   uuid.UUID           `json:"tenant_id"`
	AddressID  uuid.UUID           `json:"address_id"`
	Deposit    float64             `json:"deposit"`
	Rent       float64             `json:"rent"`
	Business   string              `json:"business"`
	StartDate  time.Time           `json:"start_date"`
	EndDate    time.Time           `json:"end_date"`
	Status     string              `json:"status"`
	Type       string              `json:"type"`
	CreatedAt  time.Time           `json:"created_at"`
	Tenant     TenantResponse      `json:"tenant"`
	Address    AddressResponse     `json:"address"`
	Guarantors []GuarantorResponse `json:"guarantors"`
}
